README.md

```diff
@@ -1,5 +1,127 @@
 # DogStatsD.Interceptors
 
+A castle proxy interceptor that branches out in to multiple interceptors to push metrics back to DataDog via DogStatD.
+
+[![install from nuget](http://img.shields.io/nuget/v/DogStatsD.Interceptors.svg?style=flat-square)](https://www.nuget.org/packages/DogStatsD.Interceptors)
+[![downloads](http://img.shields.io/nuget/dt/DogStatsD.Interceptors.svg?style=flat-square)](https://www.nuget.org/packages/DogStatsD.Interceptors)
+[![Build status](https://ci.appveyor.com/api/projects/status/7c3jrhlnp0bvph95/branch/master?svg=true)](https://ci.appveyor.com/project/Liberis/dogstatsd-interceptors/branch/master)
+
+## Getting Started
+
+DogStatsD.Interceptors is split in to 2 packages, `DogStatsD.Interceptors` and `DogStatsD.Interceptors.Annotations`. `DogStatsD.Interceptors` is the core implemtation while `DogStatsD.Interceptors.Annotations` containing only Attributes which act as method markers.
+
+DogStatsD.Interceptors can be installed via the package manager console by executing the following commandlet:
+
+```powershell
+PM> Install-Package DogStatsD.Interceptors
+```
+
+Once we have the package installed, we need to setup the DogStatsD client for c#, this is achieved at start of your application by calling `Configure` on `DogStatsd` object:
+
+```chsarp
+// The code is located under the StatsdClient namespace
+using StatsdClient;
+
+// ...
+
+var dogstatsdConfig = new StatsdConfig
+{
+    StatsdServerName = "127.0.0.1",
+    StatsdPort = 8125, // Optional; default is 8125
+    Prefix = "myApp" // Optional; by default no prefix will be prepended
+};
+
+StatsdClient.DogStatsd.Configure(dogstatsdConfig);
+```
+
+For more information on configuring DogStatD client see the [DataDogs GitHub Page](https://github.com/DataDog/dogstatsd-csharp-client).
+
+DogStatsD.Interceptors can then be plumbed in to any IoC container of choice.
+
+### Registering Interceptors
+
+#### Autofac
+
+Within the container builder within your application you will need to register the `DogStatdInterceptor`:
```

Comment:
"The `DogStatdInterceptor` will need to be registered within the application's container builder"